File: Loenn/entities/warpPedestal.lua
local pedestal = {}

pedestal.name = "CollabUtils2/WarpPedestal"
pedestal.depth = 1000
pedestal.texture = "CollabUtils2/placeholderorb/placeholderorb00"
pedestal.justification = { 0.5, 0.95 }

pedestal.placements = {
    name = "pedestal",
    data = {
        map = "Celeste/1-ForsakenCity",
        sprite = "CollabUtils2_placeholderOrb",
        returnToLobbyMode = "SetReturnToHere",
        allowSaving = true,
        fillSoundEffect = "",
        bubbleOffsetY = 16
    }
}

pedestal.fieldInformation = {
    bubbleOffsetY = {
        fieldType = "integer"
    },
    returnToLobbyMode = {
        options = { "SetReturnToHere", "RemoveReturn", "DoNotChangeReturn" },
        editable = false
    }
}

return pedestal

File: Ahorn/entities/warpPedestal.jl
module CollabUtils2WarpPedestal

using ..Ahorn, Maple

@mapdef Entity "CollabUtils2/WarpPedestal" WarpPedestal(x::Integer, y::Integer, map::String="Celeste/1-ForsakenCity", sprite::String="CollabUtils2_placeholderOrb",
    returnToLobbyMode::String="SetReturnToHere", allowSaving::Bool=true, fillSoundEffect::String="", bubbleOffsetY::Integer=16)

const placements = Ahorn.PlacementDict(
    "Warp Pedestal (Collab Utils 2)" => Ahorn.EntityPlacement(
        WarpPedestal
    )
)

Ahorn.editingOptions(entity::WarpPedestal) = Dict{String, Any}(
    "returnToLobbyMode" => String["SetReturnToHere", "RemoveReturn", "DoNotChangeReturn"]
)

function Ahorn.selection(entity::WarpPedestal)
    x, y = Ahorn.position(entity)

    return Ahorn.getSpriteRectangle("CollabUtils2/placeholderorb/placeholderorb00", x, y, jx=0.5, jy=0.92)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::WarpPedestal)
    Ahorn.drawSprite(ctx, "CollabUtils2/placeholderorb/placeholderorb00", 0, 0, jx=0.5, jy=0.92)
end

end
